// intl-tel-input-ng/esm2015/lib/intl-tel-input-ng.module.js
/*
 * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
 * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS
 * FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR
 * COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER
 * IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN
 * CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
 */
import { CommonModule } from '@angular/common';
import { NgModule } from '@angular/core';
import { FormsModule } from '@angular/forms';
import { IntlTelInputComponent } from './components/intl-tel-input.component';
export class IntlTelInputNgModule {
    static forRoot() {
        return {
            ngModule: IntlTelInputNgModule,
            providers: []
        };
    }
}
IntlTelInputNgModule.decorators = [
    { type: NgModule, args: [{
                declarations: [
                    IntlTelInputComponent,
                ],
                imports: [
                    CommonModule,
                    FormsModule,
                ],
                exports: [
                    IntlTelInputComponent,
                ]
            },] }
];
